giraph-core/src/main/java/org/apache/giraph/bsp/BspService.java

```diff
@@ -272,7 +273,7 @@ public BspService(
     }
     if (LOG.isInfoEnabled()) {
       LOG.info("BspService: Connecting to ZooKeeper with job " + jobId +
-          ", " + getTaskPartition() + " on " + serverPortList);
+          ", " + getTaskId() + " on " + serverPortList);
```

Comment:
We can't actually set `taskId` before creating the ZK client, since we need ZK to get the get the application attempt. I changed this to log the partition instead, which we can get from `conf`.